File: backend/src/main/java/cz/promtply/backend/entity/ClientToken.java
package cz.promtply.backend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.SQLUpdate;
import org.hibernate.annotations.Where;

import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
@Entity
@Table(name = "clients_tokens")
@SQLDelete(sql = "UPDATE clients_tokens SET deleted_on = CURRENT_TIMESTAMP WHERE id = ?")
@Where(clause = "deleted_on IS NULL")
public class ClientToken {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id", nullable = false)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "client_id", nullable = false)
    private Client client;

    @Column(name = "token_hash", length = 128)
    private String tokenHash;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @ColumnDefault("now()")
    @Column(name = "created_on", nullable = false)
    private Instant createdOn;

    @Column(name = "deleted_on")
    private Instant deletedOn;

    @Column(name = "expiration")
    private Instant expiration;

}

File: frontend/src/lib/types/dtos/client-token.ts
import type { UserBasicsResponse } from "./user";

export interface ClientTokenRequest {
  name: string;
  description: string;
  expiration?: string; // Date -> ISO string
}

export interface ClientTokenResponse {
  id: string; // UUID
  name: string;
  description: string;
  expiration: string | null; // Date -> ISO string
  isExpired: boolean;
  createdBy: UserBasicsResponse;
  deletedBy: UserBasicsResponse;
  lastUsedOn: string | null; // Instant
  createdOn: string; // Instant
  deletedOn: string | null; // Instant
}

export interface ClientTokenValueResponse {
  token: string;
}
